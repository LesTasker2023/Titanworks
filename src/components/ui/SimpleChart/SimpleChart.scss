@import "../../../styles/globals.scss";

// BarChart component
.simple-chart {
  &__bar-container {
    display: flex;
    flex-direction: column;
    gap: var(--space-2);
  }

  &__bar-item {
    display: flex;
    flex-direction: column;
    gap: var(--space-1);
  }

  &__bar-header {
    display: flex;
    justify-content: space-between;
    font-size: var(--font-size-xs);
    line-height: var(--line-height-xs);
  }

  &__bar-label {
    color: inherit;
  }

  &__bar-value {
    font-family: var(--font-family-mono);
  }

  &__bar-track {
    width: 100%;
    background-color: var(--color-gray-200);
    border-radius: var(--border-radius-full);
    height: var(--space-2);

    [data-theme="dark"] & {
      background-color: var(--color-gray-700);
    }
  }

  &__bar-progress {
    height: var(--space-2);
    border-radius: var(--border-radius-full);
  }

  // PieChart component
  &__pie-container {
    display: flex;
    flex-direction: column;
    gap: var(--space-2);
  }

  &__pie-item {
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: var(--font-size-sm);
    line-height: var(--line-height-sm);
  }

  &__pie-label-group {
    display: flex;
    align-items: center;
    gap: var(--space-2);
  }

  &__pie-indicator {
    width: var(--space-3);
    height: var(--space-3);
    border-radius: var(--border-radius-full);
  }

  &__pie-label {
    color: inherit;
  }

  &__pie-value {
    font-family: var(--font-family-mono);
    font-weight: var(--font-weight-medium);
  }

  // LineChart component
  &__line-container {
    display: flex;
    flex-direction: column;
    gap: var(--space-1);
  }

  &__line-item {
    display: flex;
    justify-content: space-between;
    font-size: var(--font-size-sm);
    line-height: var(--line-height-sm);
  }

  &__line-label {
    color: inherit;
  }

  &__line-value {
    font-family: var(--font-family-mono);
  }
}
