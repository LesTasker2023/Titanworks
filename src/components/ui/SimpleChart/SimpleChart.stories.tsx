import type { Meta, StoryObj } from '@storybook/nextjs-vite';
import { SimpleChart } from './SimpleChart';
import './SimpleChart.scss';

const meta: Meta<typeof SimpleChart> = {
  title: 'UI/SimpleChart',
  component: SimpleChart,
  parameters: {
    layout: 'centered',
    docs: {
      description: {
        component:
          'A lightweight chart component for simple data visualization without external dependencies.',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    type: {
      control: 'select',
      options: ['bar', 'pie', 'line'],
      description: 'The type of chart to render',
    },
    className: {
      control: 'text',
      description: 'Additional CSS classes',
    },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

// Sample data
const sampleData = [
  { name: 'Components', value: 172, color: '#3b82f6' },
  { name: 'Tests', value: 1097, color: '#10b981' },
  { name: 'Stories', value: 49, color: '#f59e0b' },
  { name: 'Files', value: 314, color: '#8b5cf6' },
];

const performanceData = [
  { name: 'Build Time', value: 84, color: '#ef4444' },
  { name: 'Bundle Size', value: 2400, color: '#f97316' },
  { name: 'Load Speed', value: 95, color: '#10b981' },
  { name: 'Lighthouse', value: 88, color: '#3b82f6' },
];

const deploymentData = [
  { name: 'Success', value: 94, color: '#10b981' },
  { name: 'Failed', value: 6, color: '#ef4444' },
];

export const BarChart: Story = {
  args: {
    data: sampleData,
    type: 'bar',
  },
};

export const PieChart: Story = {
  args: {
    data: deploymentData,
    type: 'pie',
  },
};

export const LineChart: Story = {
  args: {
    data: performanceData,
    type: 'line',
  },
};

export const PerformanceMetrics: Story = {
  args: {
    data: performanceData,
    type: 'bar',
    className: 'w-96',
  },
  parameters: {
    docs: {
      description: {
        story: 'Performance metrics visualization with custom styling.',
      },
    },
  },
};

export const EmptyData: Story = {
  args: {
    data: [],
    type: 'bar',
  },
  parameters: {
    docs: {
      description: {
        story: 'Chart handles empty data gracefully.',
      },
    },
  },
};

export const SingleDataPoint: Story = {
  args: {
    data: [{ name: 'Single Item', value: 100, color: '#8b5cf6' }],
    type: 'bar',
  },
  parameters: {
    docs: {
      description: {
        story: 'Chart with a single data point.',
      },
    },
  },
};
