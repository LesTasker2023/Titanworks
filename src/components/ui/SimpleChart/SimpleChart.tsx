// EMERGENCY: Create a simple chart replacement to eliminate Recharts bloat
// This replaces complex charts with simple visual elements for Vercel 250MB limit

import './SimpleChart.scss';

export const SimpleChart = ({
  data,
  type = 'bar',
  className = '',
}: {
  data: Array<{ name: string; value: number; color?: string }>;
  type?: 'bar' | 'pie' | 'line';
  className?: string;
}) => {
  const maxValue = Math.max(...data.map(d => d.value));

  if (type === 'bar') {
    return (
      <div className={`simple-chart__bar-container ${className}`}>
        {data.map((item, index) => (
          <div key={index} className="simple-chart__bar-item">
            <div className="simple-chart__bar-header">
              <span className="simple-chart__bar-label">{item.name}</span>
              <span className="simple-chart__bar-value">{item.value}</span>
            </div>
            <div className="simple-chart__bar-track">
              <div
                className="simple-chart__bar-progress"
                style={{
                  width: `${(item.value / maxValue) * 100}%`,
                  backgroundColor: item.color || 'hsl(var(--surface-interactive))',
                }}
              />
            </div>
          </div>
        ))}
      </div>
    );
  }

  if (type === 'pie') {
    const total = data.reduce((sum, item) => sum + item.value, 0);
    return (
      <div className={`simple-chart__pie-container ${className}`}>
        {data.map((item, index) => (
          <div key={index} className="simple-chart__pie-item">
            <div className="simple-chart__pie-label-group">
              <div
                className="simple-chart__pie-indicator"
                style={{ backgroundColor: item.color || `hsl(${index * 60}, 70%, 50%)` }}
              />
              <span className="simple-chart__pie-label">{item.name}</span>
            </div>
            <span className="simple-chart__pie-value">
              {item.value} ({Math.round((item.value / total) * 100)}%)
            </span>
          </div>
        ))}
      </div>
    );
  }

  // Default to simple list for line charts
  return (
    <div className={`simple-chart__line-container ${className}`}>
      {data.map((item, index) => (
        <div key={index} className="simple-chart__line-item">
          <span className="simple-chart__line-label">{item.name}</span>
          <span className="simple-chart__line-value">{item.value}</span>
        </div>
      ))}
    </div>
  );
};
