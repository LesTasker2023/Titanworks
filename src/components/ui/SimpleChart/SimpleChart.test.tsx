import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { SimpleChart } from './SimpleChart';

// Test data
const defaultData = [
  { name: 'Jan', value: 400 },
  { name: 'Feb', value: 300 },
  { name: 'Mar', value: 500 },
  { name: 'Apr', value: 280 },
  { name: 'May', value: 390 },
];

const renderBasicChart = (props = {}) => {
  return render(<SimpleChart data={defaultData} {...props} />);
};

describe('SimpleChart', () => {
  describe('Snapshots', () => {
    it('matches default snapshot', () => {
      const { container } = renderBasicChart();
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches bar type snapshot', () => {
      const { container } = renderBasicChart({ type: 'bar' });
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches pie type snapshot', () => {
      const { container } = renderBasicChart({ type: 'pie' });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Basic Functionality', () => {
    it('renders correctly', () => {
      renderBasicChart();
      expect(screen.getByText('Jan')).toBeInTheDocument();
      expect(screen.getByText('400')).toBeInTheDocument();
    });

    it('displays all data items', () => {
      renderBasicChart();
      defaultData.forEach(item => {
        expect(screen.getByText(item.name)).toBeInTheDocument();
        expect(screen.getByText(item.value.toString())).toBeInTheDocument();
      });
    });
  });

  describe('Chart Types', () => {
    it('renders bar chart by default', () => {
      const { container } = renderBasicChart();
      expect(container.querySelector('.simple-chart__bar-container')).toBeInTheDocument();
    });

    it('renders bar chart when type is bar', () => {
      const { container } = renderBasicChart({ type: 'bar' });
      expect(container.querySelector('.simple-chart__bar-container')).toBeInTheDocument();
    });

    it('renders pie chart when type is pie', () => {
      const { container } = renderBasicChart({ type: 'pie' });
      expect(container.querySelector('.simple-chart__pie-indicator')).toBeInTheDocument();
    });

    it('renders line chart when type is line', () => {
      renderBasicChart({ type: 'line' });
      // Line charts fall back to simple data list format
      expect(screen.getByText('Jan')).toBeInTheDocument();
      expect(screen.getByText('400')).toBeInTheDocument();
    });
  });

  describe('Props', () => {
    it('handles className prop correctly', () => {
      const { container } = renderBasicChart({ className: 'custom-chart' });
      expect(container.firstChild).toHaveClass('custom-chart');
    });

    it('handles custom colors in data', () => {
      const colorData = [
        { name: 'Red', value: 100, color: '#ff0000' },
        { name: 'Blue', value: 200, color: '#0000ff' },
      ];
      const { container } = renderBasicChart({ data: colorData });
      expect(
        container.querySelector('[style*="background-color: rgb(255, 0, 0)"]')
      ).toBeInTheDocument();
    });
  });

  describe('Edge Cases', () => {
    it('handles empty data gracefully', () => {
      const { container } = renderBasicChart({ data: [] });
      expect(container.firstChild).toBeInTheDocument();
    });

    it('handles single data item', () => {
      const singleData = [{ name: 'Single', value: 100 }];
      renderBasicChart({ data: singleData });
      expect(screen.getByText('Single')).toBeInTheDocument();
      expect(screen.getByText('100')).toBeInTheDocument();
    });

    it('handles zero values', () => {
      const zeroData = [
        { name: 'Zero', value: 0 },
        { name: 'Positive', value: 100 },
      ];
      renderBasicChart({ data: zeroData });
      expect(screen.getByText('Zero')).toBeInTheDocument();
      expect(screen.getByText('0')).toBeInTheDocument();
    });

    it('handles component unmounting cleanly', () => {
      const { unmount } = renderBasicChart();
      expect(() => unmount()).not.toThrow();
    });

    it('preserves functionality after remounting', () => {
      const { unmount } = renderBasicChart();
      unmount();

      renderBasicChart();
      expect(screen.getByText('Jan')).toBeInTheDocument();
    });
  });
});
