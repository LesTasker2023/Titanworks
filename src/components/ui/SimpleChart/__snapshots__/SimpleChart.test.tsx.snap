// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SimpleChart > Snapshots > matches bar type snapshot 1`] = `
<div
  class="simple-chart__bar-container "
>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        Jan
      </span>
      <span
        class="simple-chart__bar-value"
      >
        400
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 80%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        Feb
      </span>
      <span
        class="simple-chart__bar-value"
      >
        300
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 60%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        Mar
      </span>
      <span
        class="simple-chart__bar-value"
      >
        500
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 100%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        Apr
      </span>
      <span
        class="simple-chart__bar-value"
      >
        280
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 56.00000000000001%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        May
      </span>
      <span
        class="simple-chart__bar-value"
      >
        390
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 78%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
</div>
`;

exports[`SimpleChart > Snapshots > matches default snapshot 1`] = `
<div
  class="simple-chart__bar-container "
>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        Jan
      </span>
      <span
        class="simple-chart__bar-value"
      >
        400
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 80%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        Feb
      </span>
      <span
        class="simple-chart__bar-value"
      >
        300
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 60%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        Mar
      </span>
      <span
        class="simple-chart__bar-value"
      >
        500
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 100%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        Apr
      </span>
      <span
        class="simple-chart__bar-value"
      >
        280
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 56.00000000000001%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
  <div
    class="simple-chart__bar-item"
  >
    <div
      class="simple-chart__bar-header"
    >
      <span
        class="simple-chart__bar-label"
      >
        May
      </span>
      <span
        class="simple-chart__bar-value"
      >
        390
      </span>
    </div>
    <div
      class="simple-chart__bar-track"
    >
      <div
        class="simple-chart__bar-progress"
        style="width: 78%; background-color: hsl(var(--surface-interactive));"
      />
    </div>
  </div>
</div>
`;

exports[`SimpleChart > Snapshots > matches pie type snapshot 1`] = `
<div
  class="simple-chart__pie-container "
>
  <div
    class="simple-chart__pie-item"
  >
    <div
      class="simple-chart__pie-label-group"
    >
      <div
        class="simple-chart__pie-indicator"
        style="background-color: rgb(217, 38, 38);"
      />
      <span
        class="simple-chart__pie-label"
      >
        Jan
      </span>
    </div>
    <span
      class="simple-chart__pie-value"
    >
      400
       (
      21
      %)
    </span>
  </div>
  <div
    class="simple-chart__pie-item"
  >
    <div
      class="simple-chart__pie-label-group"
    >
      <div
        class="simple-chart__pie-indicator"
        style="background-color: rgb(217, 217, 38);"
      />
      <span
        class="simple-chart__pie-label"
      >
        Feb
      </span>
    </div>
    <span
      class="simple-chart__pie-value"
    >
      300
       (
      16
      %)
    </span>
  </div>
  <div
    class="simple-chart__pie-item"
  >
    <div
      class="simple-chart__pie-label-group"
    >
      <div
        class="simple-chart__pie-indicator"
        style="background-color: rgb(38, 217, 38);"
      />
      <span
        class="simple-chart__pie-label"
      >
        Mar
      </span>
    </div>
    <span
      class="simple-chart__pie-value"
    >
      500
       (
      27
      %)
    </span>
  </div>
  <div
    class="simple-chart__pie-item"
  >
    <div
      class="simple-chart__pie-label-group"
    >
      <div
        class="simple-chart__pie-indicator"
        style="background-color: rgb(38, 217, 217);"
      />
      <span
        class="simple-chart__pie-label"
      >
        Apr
      </span>
    </div>
    <span
      class="simple-chart__pie-value"
    >
      280
       (
      15
      %)
    </span>
  </div>
  <div
    class="simple-chart__pie-item"
  >
    <div
      class="simple-chart__pie-label-group"
    >
      <div
        class="simple-chart__pie-indicator"
        style="background-color: rgb(38, 38, 217);"
      />
      <span
        class="simple-chart__pie-label"
      >
        May
      </span>
    </div>
    <span
      class="simple-chart__pie-value"
    >
      390
       (
      21
      %)
    </span>
  </div>
</div>
`;
